Kanan/Currtarget.cpp

```diff
@@ -0,0 +1,256 @@
+#pragma once
+#define IMGUI_DEFINE_MATH_OPERATORS
+#include <imgui.h>
+#include <imgui_internal.h>
+
+#include "Mabinogi.hpp"
+#include "KCharacter.hpp"
+#include "Kanan.hpp"
+#include "Currtarget.hpp"
+
+namespace kanan {
+
+	//most of this is based off status ui and entity viewer
+
+
+
+	Currtarget::Currtarget()
+		: m_hp{}
+	{
+		m_hp.reserve(100);
+	}
+
+	void Currtarget::onFrame()
+	{
+
+		auto game = g_kanan->getGame();
+		auto localCharacter = game->getLocalCharacter();
+		KCharacter* target = nullptr;
+
+		//check if there is a local char or target for local char. if not dont bother rendering window
+		if (!localCharacter) { hastarget = false; }
+		else { hastarget = true; }
+		if (localCharacter) {
+			target = g_kanan->getGame()->getCharacterByID(localCharacter->targetID);
+			if (target == nullptr) { hastarget = false; }
+			else { hastarget = true; }
+		}
```

Comment:
Could be replaced with clearer logic. Maybe something like:
```
hastarget = localCharacter && localCharacter->targetID != 0;
```

Reply:
not sure how to do this cleaner, i only want to checking for target if local player exists